### DISCUSSION TITLE
Integration API

### DISCUSSION THREAD

**Discussion by me-matt** [foambubble/foam#1006] [General]
There are a few other packages that aim to enhance the features available in foam, e.g. KaleidoFoam.

Has there been any consideration for an integration API? Something enabling developers to add features or extend the behaviour of existing features.

Considering how foam is using lerna, I think this question may also relate to how core features scale in the main package. Over time, is the intent to refactor existing features into sibling packages of foam-vscode?

I don't have any particular use case here, just trying to develop a better understanding of the long-term intent for foams design 🙂

**Comment by riccardoferretti** [comment]
Hi @me-matt - yes the idea is for Foam to have an API that other extensions can use to integrate with the knowledge base.
Given the backward compatibility issues involved, we wanted to have a more "stable" model first, hopefully the API will be released soon though.
